### DISCUSSION TITLE
extensions/v1beta1 Ingress deprecation

### DISCUSSION THREAD

**Discussion by MattPOlson** [argoproj/argo-cd#8121] [Q&A] [answered]
Even though we don't have any ingress configured on our cluster I still see calls to the deprecated ingress api 

`2742a7d9717f","stage":"ResponseStarted","requestURI":"/apis/extensions/v1beta1/ingresses?allowWatchBookmarks=true\u0026resourceVersion=438774502\u0026watch=true","verb":"watch","user":{"username":"system:serviceaccount:kube-system:argocd-manager","uid":"e2183625-d2f3-44a6-80cd-8b6f1f4bad46`

And I see this in the controller logs 

`W0107 06:50:23.175550       1 warnings.go:70] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
W0107 06:50:23.190150       1 warnings.go:70] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress`

I can't figure out what's calling that and when we upgrade the cluster to v1.22 are we going to have an issue?

**Comment by alexmt** [comment, marked as answer]
Hello @MattPOlson ,

I just run into this message as well. This warning message is returned by K8S API server and printed by the k8s client. The warning is printed because Argo CD is trying to monitor all available K8S resources including "extensions". After migrating to v1.22 warning message will disappear. There should be no issues as long as you don't have "extensions" resources in Git.

**Reply by MattPOlson** [reply]
Awesome, thanks for the help!

**Reply by insider89** [reply]
We are using GKE v1.22.10-gke.600 and ArgoCD v2.3.3. Still see this message in the log.

**Comment by nhuray** [comment]
Hi @alexmt 

I have the same issue in GKE 1.21. The message is saying the cluster can't be scaled to 1.22 because`argocd-application-controller` is using the deprecated `v1beta1/ingress` API:

![Screen Shot 2022-07-28 at 10 30 23 AM](https://user-images.githubusercontent.com/958983/181464009-b4b7deed-8e33-41d1-b8d1-7c4b6773f05b.png)

I'm running ArgoCD version v2.4.7+81630e6.

Is there's a way to fix that issue ?

**Reply by castaneai** [reply]
Presumably, this can be safely ignored.
This issue comment may be helpful: https://github.com/argoproj/argo-cd/issues/9786#issuecomment-1179429531

**Comment by mortya** [comment]
It would be really nice if there were a configuration mechanism to prevent argocd from accessing this API, so we can be certain that there are no hidden dependencies here.  I've used resourceExclusions / resource.exclusions before.  Like so:
```
  resource.exclusions: |
    - apiGroups:
      - ingresses.v1beta1.extensions
```
But it didn't help here.  Am I doing something wrong?  Is there some other way to do this?  Or should I submit an issue?

(Update: went ahead and submitted a new question, https://github.com/argoproj/argo-cd/discussions/13259 .  Probably should not have piggybacked on this existing question.)
